### PR TITLE
TE-798 Type into implemented

### PR DESCRIPTION
**Pull request by kbayraktar**

### CODE REVIEW

**Comment by gunther-bachmann** [comment]
DO NOT MERGE until feature/status is merged into develop. then merge into develop

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: What exactly is tested here?

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: This will probably fail on the build server, wouldn't it?

Comment (gunther-bachmann) [reply src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: why not have somthing like assumeTrue(hostDriverFixture.connected()); within the tests and check availibility of env parameters before actually attempting the connect here?

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: why not log this out if the output is of interest?

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: Wouldn't 'typeAt' be more appropriate?

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.aml
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.aml]: Wouldn't 'typeAt' be more appropriate?

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.aml
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.aml]: Syntax conventions suggest "Get status"

File: src/main/java/org/testeditor/fixture/host/net/Connection.java
Comment (gunther-bachmann) [inline src/main/java/org/testeditor/fixture/host/net/Connection.java]: Aside from this we should talk about logging conventions in the team (syntax, level ...)

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: hostUrl should be HOST_URL, right?

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: This should be S3270_PATH (typo)

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: Should be HOST_PORT, right?

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (gunther-bachmann) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: why not name it 'hostDriverFixture', we don't use this variable too much, so the abbreviation does not help, does it?

**Comment by gunther-bachmann** [comment]
By the way, is this TE issue number 798 really correct? Shouldn't it be 498?

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (franzbecker) [reply src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: No it's not a constant.

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (franzbecker) [reply src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: Should be s3270Path as it's not a constant.

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: typo: should be "and"

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: Is there any way we can prevent that?

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: the comment indicates that the time unit is confusing. Instead of the comment the parameter should be called `milliseconds`.

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: Why? 😕

File: src/main/java/org/testeditor/fixture/host/net/Connection.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/net/Connection.java]: "successful" with 2 s

File: src/main/java/org/testeditor/fixture/host/net/HelperTool.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/net/HelperTool.java]: The name of this class is to generic

File: src/main/java/org/testeditor/fixture/host/net/HelperTool.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/net/HelperTool.java]: That's a pretty mean side-effect. You ask a question - is this path valid - and then change the internal static state 💣

File: src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java
Comment (franzbecker) [inline src/test/java/org/testeditor/fixture/host/HostDriverFixtureTest.java]: constants should be static and final

File: src/main/java/org/testeditor/fixture/host/net/HelperTool.java
Comment (franzbecker) [inline src/main/java/org/testeditor/fixture/host/net/HelperTool.java]: There is already a  utility function for that. Why add it here? As I see it, it is only used in the test. No need to add it to production code then.

**Review by franzbecker — CHANGES_REQUESTED** [review]
Almost there, just a few minor changes

File: src/main/java/org/testeditor/fixture/host/HostDriverFixture.java
Comment (kbayraktar) [reply src/main/java/org/testeditor/fixture/host/HostDriverFixture.java]: Not yet, therefore we need to have more experience with the s2370 driver.

**Pull request merged**
